test/e2e/fleet_test.go

```diff
@@ -294,11 +293,14 @@ func TestFleetRollingUpdate(t *testing.T) {
 					maxUnavailable, err := intstr.GetValueFromIntOrPercent(flt.Spec.Strategy.RollingUpdate.MaxUnavailable, 100, true)
 					assert.Nil(t, err)
 					target := float64(targetScale)
-					if len(list.Items) > int(target+math.Ceil(target*float64(maxSurge)/100.)+math.Ceil(target*float64(maxUnavailable)/100.)) {
-						err = errors.New("New replicas should be less than target + maxSurge + maxUnavailable")
-					}
-					if err != nil {
-						return false, err
+
+					if !runtime.FeatureEnabled(runtime.FeatureRollingUpdateOnReady) {
+						if len(list.Items) > int(target+math.Ceil(target*float64(maxSurge)/100.)+math.Ceil(target*float64(maxUnavailable)/100.)) {
+							err = errors.New(fmt.Sprintf("New replicas should be less than target + maxSurge + maxUnavailable %d %d", len(list.Items), int(target+math.Ceil(target*float64(maxSurge)/100.)+math.Ceil(target*float64(maxUnavailable)/100.))))
+						}
+						if err != nil {
+							return false, err
+						}
```

Comment:
Please, put `int(target+math.Ceil(target*float64(maxSurge)/100.)+math.Ceil(target*float64(maxUnavailable)/100.))` in a separate variable and reuse it in _if_ clause and in log message. Also you can return an error immediatelly, ommiting `if err != nil`.